lib/matplotlib/backends/qt_compat.py

```diff
@@ -109,9 +109,18 @@
         except:
             res = 'QVariant API v2 specification failed. Defaulting to v1.'
             verbose.report(cond + res, 'helpful')
+    if QT_API == QT_API_PYQT5:
+        try:
+            from PyQt5 import QtCore, QtGui, QtWidgets
+            _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
+        except ImportError:
+            # fell through, tried PyQt5, failed fall back to PyQt4
+            QT_API = rcParams['backend.qt4']
```

Comment:
Perhaps I misunderstand, but as far as I can see that will only happen if 

* You have specified `rcParams['backend.qt4'] = PyQt5` which is clearly wrong.
* Tried using the Qt5Agg backend without installing PyQt5
* if `if "PyQt4" in sys.modules or "PySide" in sys.modules:` fails if which case this will fail any way.